drag_and_drop_v2/public/js/drag_and_drop.js

```diff
@@ -860,6 +859,44 @@ function DragAndDropBlock(runtime, element, configuration) {
         });
     };
 
+    var doAttempt = function(evt) {
+        evt.preventDefault();
+
+        $.ajax({
+            type: 'POST',
+            url: runtime.handlerUrl(element, "do_attempt"),
+            data: '{}'
+        }).done(function(data){
+            state.num_attempts = data.num_attempts;
+            state.overall_feedback = data.feedback;
+            if (data.attempts_remain) {
+                for (var i=0; i < data.misplaced_items.length; i++) {
```

Comment:
Nit: We usually prefer `data.misplaced_items.forEach(function(misplaced_item_id) { ... })`.